agent/csi/plugin.go

```diff
@@ -88,16 +140,23 @@ func (np *NodePlugin) GetPublishedPath(volumeID string) string {
 }
 
 func (np *NodePlugin) NodeGetInfo(ctx context.Context) (*api.NodeCSIInfo, error) {
+	np.mu.RLock()
+	defer np.mu.RUnlock()
 
-	resp := &csi.NodeGetInfoResponse{
-		NodeId: np.nodeID,
+	resp, err := np.Client().NodeGetInfo(ctx, &csi.NodeGetInfoRequest{})
+	if err != nil {
+		return nil, err
 	}
 
 	return makeNodeInfo(resp), nil
 }
 
 func (np *NodePlugin) NodeStageVolume(ctx context.Context, req *api.VolumeAssignment) error {
 
+	if !np.staging {
+		return status.Error(codes.Unimplemented, "Plugin doesn't support Stage")
```

Comment:
Don't return an error here. Instead, if staging isn't supported, return nil. The caller doesn't care if staging is supported or not, it just needs to know the volume is staged if it can be.